__test__/integrationTests.test.ts

```diff
@@ -120,6 +156,35 @@ describe("Integration tests for AUTH routes:", () => {
     // Reject invalid inputs
     // Reject unexpected attributes
     // Others
+    // user validation test
+    describe("user validation test : ", () => {
+        test.each([
+            { missingFieldName: "userName" },
+            { missingFieldName: "email" },
+            { missingFieldName: "password" },
+        ])(
+            "return 400 when $missingFieldName field is missing",
+            async ({ missingFieldName }) => {
+                const user: RegistrationData = createRandomUserForRegister();
+
+                delete user[missingFieldName as keyof typeof user];
+
+                const response = await axios.post("api/auth/register", user);
+
+                expect(response.status).toBe(400);
+            }
+        );
+
+        test("return 400 when password is too short", async () => {
```

Comment:
This is good, but there will need to be more tests to make sure the regex pattern works correctly. Eg. password too long, no special character, no capital letters, no lowercase letters, no numbers etc...
email tests for a number of possibly invalid emails
username tests length and data type, test with numbers, short, long etc.